open-authoring/open-authoring/OpenAuthoringProvider.tsx

```diff
@@ -113,55 +101,83 @@ export const OpenAuthoringProvider = ({
   )
 }
 
-const OpenAuthoringAuthModal = ({ onUpdateAuthState, close }) => {
-  let modalProps
+const OpenAuthoringAuthModal = ({ close }) => {
+  const [modalProps, setModalProps] = useState({
+    title: 'Loading...',
+    message:
+      '',
+    actions: [
+      {
+        name: 'Cancel',
+        action: close,
+      },
+      {
+        name: 'Loading...',
+        action: async () => {
+          
+        },
+        primary: true,
+      },
+    ],
+  })
 
   const openAuthoring = useOpenAuthoring()
   const cms = useCMS()
 
-  if (!openAuthoring.authenticated) {
-    modalProps = {
-      title: 'GitHub Authorization',
-      message:
-        'To save edits, Tina requires GitHub authorization. On save, changes will get commited to GitHub using your account.',
-      actions: [
-        {
-          name: 'Cancel',
-          action: close,
-        },
-        {
-          name: 'Continue to GitHub',
-          action: async () => {
-            await openAuthoring.authenticate()
-            onUpdateAuthState()
+  openAuthoring.userIsAuthenticated().then( isAuthenticated => {
```

Comment:
I believe this will make a request every time this modal rerenders

Reply:
I think this could maybe be reorganized a bit. We already made a request to see if we are authorized and if the fork exists, to get to the point where we're showing the modal. Can we pass those values through as props? Also, then I don't think we would even need a loading state